### ISSUE TITLE
DenoiseDat.m needs proper testing

### ISSUE THREAD

**Issue by ryanharvey1** [ayalab1/neurocode#11]
[DenoiseDat ](https://github.com/ayalab1/neurocode/blob/master/preProcessing/DenoiseDat.m) was pushed to run by default f69a1ace33e0c769be1b7887d149b2bd383d204c back in Feb. but was never fully tested. It was removed from default in commit d8acfaf71f5f60141adc3a078e7800a5b35a83c0 . 

There has been reports that it removes spikes and distorts signal during common instances such as SWRs and opto pulses. Without full testing, it's hard to know how this function could affect your signal.

**Comment by YY535** [comment]
Can you show some examples of  SWRs and opto pulses distortion? 
In the default pipeline: preprocessSession.m , the denoising (DenoiseDat) happens at line 252 (line 244-253), which is designed to avoid distortion of LFP signals which is generated at line 198-216. 
Therefore, the denoise is specifically for spike sorting at line 255-287.

**Comment by ryanharvey1** [comment]
I've heard from @lkaraba, @heathlarsson, and @amymarple that it has negatively affected their signal or removed spikes during opto pulses, so maybe they have examples. However, it would have been good to test how this function affects these things before it was made default.

**Comment by heathlarsson** [comment]
Yes for opto the noise removal will lose units. Needs to be fully tested.

**Comment by YY535** [comment]
Can you please give explicit examples/sessions? What exactly happens there? We need to see the problem to solve it, it would be related to component selection.

**Comment by AntonioFR8** [comment]
Yes, we should try to fix this. It can be super useful and save a lot of
manual curation time

**Comment by ryanharvey1** [comment]
Example on how DenoiseDat alters raw signal. The first image is original, and the second is following DenoiseDat. Because average waveform extraction is preformed on this altered signal, sessions that used this function may likely have inaccurate cell trilateration. 

![Original signal](https://github.com/user-attachments/assets/d9bc3215-f542-49ba-ba65-aa8a8891f813)

![Altered signals](https://github.com/user-attachments/assets/332427fc-5157-40d7-a505-c592c56b6380)

**Comment by ryanharvey1** [comment]
Additionally, a very important point is that the denoising is preformed across the entire recording session and not just high emg intervals.

**Comment by AntonioFR8** [comment]
Thanks Ryan. Why would this change waveforms in this way? It seems that
waveform themselves are changed, is that the case?

**Comment by ryanharvey1** [comment]
Has there been any development on this issue?

**Comment by AntonioFR8** [comment]
I think weiwei made a newer and better version that some people are usign

**Comment by ryanharvey1** [comment]
@YY535 , you could put your updates and tests in a PR so we can review them.
